mdns: vrrp: T3635: Add ability to use mDNS repeater with VRRP

sarthurdev opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a commend and is not rendered -->

## Change Summary
<!--- Provide a general summary of your changes in the Title above -->
This allows the mDNS repeater to be enabled on VRRP-enabled routers as only the master router will repeat the packets.

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
* https://phabricator.vyos.net/T3635

## Component(s) name
<!-- A rather incomplete list of components: ethernet, wireguard, bgp, mpls, ldp, l2tp, dhcp ... -->
mdns, vrrp

## Proposed changes
<!--- Describe your changes in detail -->
When the new setting is active, mDNS repeater will disable mDNS on interfaces where VRRP is active and the state is not MASTER.
A hook is also added to keepalived_fifo to keep the mDNS repeater updated as VRRP states change.

## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```
-->
```
set service mdns repeater interface eth0
set service mdns repeater interface eth1
set service mdns repeater disable-on-vrrp
```
Assuming VRRP is enabled on eth0 and eth1, if the router is not the master router mDNS will be disabled to prevent a packet storm. If the state changes to master, mDNS will then be enabled on each master interface.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

sarthurdev committed:
mdns: vrrp: T3635: Add ability to use mDNS repeater with VRRP

c-po commented on src/conf_mode/service_mdns-repeater.py:
```diff
@@ -16,17 +16,20 @@
 airbag.enable()
 
 config_file = r'/etc/default/mdns-repeater'
+vrrp_running_file = '/run/mdns_vrrp_active'
```
Why not simply check if keepalived is running?

`vyos.util.process_named_running('keepalived')`

sarthurdev replied:
That file is only created when `disable-on-vrrp` is set and there are VRRP groups. Only if it exists will the handler call the mdns repeater conf script on state changes.

c-po commented on src/helpers/vyos-vrrp-mdns.sh:
```diff
@@ -0,0 +1,24 @@
+
+case "$3" in
+    "MASTER"|"BACKUP"|"FAULT")
+        if [ -f /run/mdns_vrrp_active ]; then
```
There are already VRRP/keepalived state transition scripts but I'm not into detail. Maybe @DmitriyEshenko knows more?

sarthurdev replied:
This handler is called by the existing transition script: https://github.com/vyos/vyos-1x/pull/887/commits/3a77a1f25aabd92a1785fcdc8c89d65a2cf8d5de#diff-33fc5560ca57968dd54276c71f0a4c5e29f2ce34ff4d937b58672e112a19c75aR127

c-po commented:
Can we name it `disable-vrrp` or `vrrp-disable`?

sarthurdev commented:
> 
> 
> Can we name it `disable-vrrp` or `vrrp-disable`?

Updated name to `vrrp-disable` so it doesn't also share autocomplete with `disable`.

Also removed the handler shell script and let keepalive_fifo call the python script directly, as it doesn't really need to go through a shell script.
